### ISSUE TITLE
As I user I would like to visualize well correlation per zone or formation for multiple selected wells to be able to see how different wells are "connected" to each other in same type of geology

### ISSUE THREAD

**Issue by tagr71** [equinor/webviz-subsurface-components#1051]
@Midtveit:
> Usecase: As a user I like to visualize well correlation per zone or formation for multiple selected wells to be able to see how different wells are "connected" to each other in same type of geology.
> - [ ] Optionally be able to use selected log "Zone" and choose values of zone (color and label and pattern if relevant) as fill between the wells
> - [ ] Optionally be able to use selected log "Zone" and choose values of zone (color and label and pattern if relevant) as fill between behind the tracks for each well.
> - [ ] Optionally be able to use selected log "Zone" and choose values and top or bottom of selected zone values as flattening
> Maximum two (2) flattening in total
> 
> NB: the above use cases are similar to what we just implemented for well picks.
> However the input is now a discrete log and not well picks.
> NB2: Creating well picks from discrete log "Zone" are trivial in simple cases. But very complicated in case of errosion, faulting and where well is first going downdip and then updip.
> NB3: @tagr71 what do you suggest visualiation to be if there are several intervals in a well which has the same "zone" log value. ALternative A: we just use all intervals of the zone log value, such a correlation view could be messy when for example a well go in and out of a zone.

**Comment by tagr71** [comment]
Mock-up:

[https://statoilsrm.sharepoint.com/:p:/s/EffectiveSubsurfaceCharacterizationESC/EZsonxNXFZJPko-uZLVaMiEBDmEXdYmwnyaNjzRz_eXQJg?e=ckATEs](url)

**Comment by hkfb** [comment]
@tagr71 Could you post a snapshot of your mockup here?

**Comment by tagr71** [comment]
Screen shots from PPT:

![image](https://user-images.githubusercontent.com/77675910/173757387-3b9bdd4e-3a06-4281-9c1a-fa28015966f8.png)

![image](https://user-images.githubusercontent.com/77675910/173757548-aaeb4061-4f6c-4b19-b222-4230921e4236.png)

![image](https://user-images.githubusercontent.com/77675910/173757604-1ab91046-a407-4d7c-8885-53cf5ba7a21e.png)

**Comment by hkfb** [comment]
@tagr71 @Midtveit Could a first step be to add well pick support to the well log viewer, and then add support for aligning across tracks on a chosen pick?

**Comment by tagr71** [comment]
> @tagr71 @Midtveit Could a first step be to add well pick support to the well log viewer, and then add support for aligning across tracks on a chosen pick?

Sorry for very late reply. Sounds reasonable to me but I am not an expert

**Comment by hkfb** [comment]
@Midtveit @tagr71 can we consider this done after the recent improvements to the well log viewer?

**Comment by Midtveit** [comment]
Hi @tagr71 
Let me rephrase the request and see if I have understood.

Usecase: As a user I like to visualize well correlation per zone or formation for multiple selected wells to be able to see how different wells are "connected" to each other in same type of geology.
Task1: Optionally be able to use selected log "Zone" and choose values of zone (color and label and pattern if relevant)  as fill between the wells 
Task2: Optionally be able to use selected log "Zone" and choose values of zone (color and label and pattern if relevant)  as fill between behind the tracks for each well. 
Taks 3: Optionally be able to use selected log "Zone" and choose values and top or bottom of selected zone values as flattening 
Maximum two (2) flattening in total 

NB: the above use cases are similar to what we just implemented for well picks. 
However  the input is now a discrete log and not well picks.
NB2: Creating well picks from discrete log "Zone" are trivial in simple cases. But very complicated in case of errosion, faulting and where well is first going downdip and then updip.
NB3: @tagr71  what do you suggest visualiation to be if there are several intervals in a well which has the same "zone" log value. ALternative A: we just use all intervals of the zone log value, such a correlation view could be messy when for example a well go in and out of a zone.
